### PR TITLE
predict method in the base, homogenising recon_loss

### PR DESCRIPTION
**Pull request by soumickmj**
Hey @clementchadebec 
It's the same as the pull request #73 
Just did those requested changes!
Let me know if I should modify something!
Thanks :)

### CODE REVIEW

File: src/pythae/models/base/base_model.py [hunk 1]
```diff
@@ -117,6 +117,26 @@ def reconstruct(self, inputs: torch.Tensor):
         """
         return self(DatasetOutput(data=inputs)).recon_x
 
+    def predict(self, inputs: BaseDataset, **kwargs) -> ModelOutput:
+        """The input data is encoded and decoded without computing loss
+        Args:
```
Comment (clementchadebec) [inline src/pythae/models/base/base_model.py:122]: Add one line space before `Args`.

File: src/pythae/models/base/base_model.py [hunk 2]
```diff
@@ -117,6 +117,26 @@ def reconstruct(self, inputs: torch.Tensor):
...
+            inputs (BaseDataset): An instance of pythae's datasets
+        Returns:
```
Comment (clementchadebec) [inline src/pythae/models/base/base_model.py:124]: Add one line space before `Returns`.

File: src/pythae/models/base/base_model.py [hunk 3]
```diff
@@ -117,6 +117,26 @@ def reconstruct(self, inputs: torch.Tensor):
         """
         return self(DatasetOutput(data=inputs)).recon_x
 
+    def predict(self, inputs: BaseDataset, **kwargs) -> ModelOutput:
+        """The input data is encoded and decoded without computing loss
+        Args:
+            inputs (BaseDataset): An instance of pythae's datasets
```
Comment (clementchadebec) [inline src/pythae/models/base/base_model.py:123]: I just realized. To make it consistent with `reconstruct` and `interpolate` maybe we should expect `torch.Tensor`  type from the `inputs`? Let me know what is best suited for your use cases.

File: src/pythae/models/base/base_model.py [hunk 4]
```diff
@@ -117,6 +117,26 @@ def reconstruct(self, inputs: torch.Tensor):
...
+        Returns:
+            ModelOutput: An instance of ModelOutput containing reconstruction and embedding
+        """
+
+        x = inputs["data"]
```
Comment (clementchadebec) [inline src/pythae/models/base/base_model.py:128]: If inputs are `torch.Tensor` this line is not needed

File: tests/test_AE.py [hunk 5]
```diff
@@ -355,6 +355,29 @@ def test_reconstruct(self, ae, demo_data):
...
+    @pytest.fixture()
+    def ae(self, model_configs, demo_data):
+        model_configs.input_dim = tuple(demo_data[0].shape)
+        return AE(model_configs)
+
+    def test_predict(self, ae, demo_data):
+
+        model_output = ae.predict(demo_data)
+        assert tuple(model_output.recon_x.shape) == demo_data.shape
+        assert tuple(model_output.embedding.shape) == ae.model_config.latent_dim.shape
```
Comment (clementchadebec) [inline tests/test_AE.py:380]: This sould be replaced with 

```python
assert tuple(model_output.embedding.shape) == (demo_data.shape[0], ae.model_config.latent_dim)
```

**Review by clementchadebec — CHANGES_REQUESTED** [review]

**Comment by clementchadebec** [comment]
Hi @soumickmj, thank you so much for making those changes in such short notice! See above some tiny issues that still need to be addressed before merging.

**Comment by clementchadebec** [comment]
@ravih18 if you are fine with those changes, my comments and agree with the following, I will merge this PR (adding you as co-author) instead of #73 since the branch is already up-to-date with `main`.

File: src/pythae/models/base/base_model.py (hunk 3 above)
Comment (soumickmj) [reply src/pythae/models/base/base_model.py:123]: Ya, good idea!
I have modified accordingly!

**Comment by soumickmj** [comment]
Hi @clementchadebec 
My pleasure!
I have modified modified those issues. Let me know if everything is fine now :)

**Review by clementchadebec — APPROVED** [review]
Thanks for making those changes! Everything looks good now :)

**Comment by ravih18** [comment]
Hello to both of you, 
Thank you for completing this PR it looks good to me !

File: src/pythae/models/base/base_model.py (hunk 3 above)
Comment (clementchadebec) [reply src/pythae/models/base/base_model.py]: Thanks!

**Review by ravih18 — APPROVED** [review]

**Comment by clementchadebec** [comment]
Thank you both of you for this contribution !

**Pull request merged**